src/agent/onefuzz-task/src/tasks/coverage/generic.rs

```diff
@@ -45,7 +47,9 @@ pub struct Config {
     pub target_options: Vec<String>,
     pub target_timeout: Option<u64>,
 
-    pub coverage_filter: Option<String>,
```

Comment:
This deletion means that, on instance update, if there are old coverage tasks that are enqueued for a pool, they will fail when agents pick them up (because the agent isn't aware of this type).

We could add it as an unused, optional field to let the task run (but any existing `coverage_filter` will be ignored).

I believe that old tasks that do NOT have a coverage filter will be unimpacted.

Reply:
@ranweiler  Is this field currently used. 
I suggest that we keep it for this release and fail the task with a message saying that option is no longer supported. 
We can remove that option on the next release. By that time, any old tasks using this would have been drained from the queue.

Reply:
Got it, so you're basically suggesting just failing more verbosely and specifically on the first release that contains the new task?

Then the first line of the task would be something like:
```rust
if config.coverage_filter.is_some() {
    bail!("the `coverage_filter` option for the `coverage` task is no longer supported");
}
```

We could also choose to ignore that field, run the task, and just _warn_ the user, but I'm don't think we can alert them reasonably.

Reply:
I assume that we agree that it'd be a breaking change / major version bump either way; this is just to make it easier to understand why the old tasks don't get run.

Reply:
I agree. This will save the investigation time before we just tell the use to just resubmit their job.